src/ui/pages/work/partusage/PartUsageDialog.js

```diff
@@ -458,23 +612,62 @@ function PartUsageDialog(props) {
                                     formData.storeCode,
                                     formData.partCode
                                 ]}
-                                onChange={createOnChangeHandler(FORM.ASSET, FORM.ASSET_DESC, null, updateFormDataProperty, handleAssetChange)}
+                                onChange={createOnChangeHandler(FORM.ASSET, FORM.ASSET_DESC, null, updateFormDataProperty,
+                                    (asset) =>
+                                        runUiBlockingFunction(
+                                            handleAssetChange.bind(null, asset)
+                                        )
+                                )}
                                 barcodeScanner
                                 renderDependencies={[formData.partCode]}
                                 errorText={errorMessages?.assetIDCode}
                             />
 
                             <EAMSelect
                                 {...processElementInfo(tabLayout['bincode'])}
-                                disabled={!formData.storeCode}
+                                disabled={
+                                    !formData.storeCode ||
+                                    !formData.activityCode ||
+                                    (isTrackedByAsset && (
+                                        formData.transactionType === issueType ||
+                                        (formData.transactionType === returnType && !formData.assetIDCode))
```

Comment:
This might be correct, but why cheking for assetIDCode only on returns?

Reply:
Because on issue, when a part is tracked by asset you already know that the bin will get auto-completed by the single possible value (the user is not meant to be able to change it, just like in extended).

On return, a case where you will be presented with more than 1 bin, while it is true that in extended you can select the bin before selecting the asset, you also get the red asterisk on the Asset ID field signifying that is must be filled. However, we cannot get this red asterisk in light because we rely on the initially loaded layout (in extended the asterisk only appears after a quantity is selected). Additionally, it made the code a bit cleaner/legible in terms of loading the bin/lot data and field clearing from what I can recall, so I thought that the previous point of better UX in light (imo) with cleaner code would make it the better choice.